tools/kprint/main.cpp

```diff
@@ -1,33 +1,36 @@
 #include "kllvm/ast/AST.h"
 #include "kllvm/printer/printer.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/WithColor.h"
 
 #include <iostream>
 #include <string>
 #include <unistd.h>
 
 using namespace kllvm;
+using namespace llvm;
+
+cl::opt<std::string> DefinitionFilename(
+    cl::Positional, cl::desc("<definition.kore>"), cl::Required);
+
+cl::opt<std::string>
+    PatternFilename(cl::Positional, cl::desc("<pattern.kore>"), cl::Required);
+
+cl::opt<std::string> FilterSubst(cl::Positional, cl::desc("[true|false]"));
+
+cl::opt<std::string> ArgColor(cl::Positional, cl::desc("[true|false|auto]"));
 
 int main(int argc, char **argv) {
-  if (argc != 3 && argc != 4 && argc != 5) {
-    std::cerr
-        << "usage: " << argv[0]
-        << " <definition.kore> <pattern.kore> [true|false|auto] [true|false]"
-        << std::endl;
-  }
-
-  bool hasColor, filterSubst;
-  if (argc >= 4) {
-    std::string arg = argv[3];
-    hasColor = arg == "true" || (arg == "auto" && isatty(1));
-  } else {
-    hasColor = isatty(1);
-  }
-  if (argc >= 5) {
-    std::string arg = argv[4];
-    filterSubst = arg == "true";
-  } else {
-    filterSubst = true;
-  }
-
-  printKORE(std::cout, argv[1], argv[2], hasColor, filterSubst);
+  cl::PrintOptionValues();
+  StringMap<cl::Option *> &Map = cl::getRegisteredOptions();
+  Map["color"]->setHiddenFlag(cl::Hidden);
+
+  cl::ParseCommandLineOptions(argc, argv);
+
+  bool hasColor = ArgColor == "true" || (ArgColor == "auto" && isatty(1))
+                  || (ArgColor == "" && isatty(1));
+  bool filterSubst = FilterSubst == "true" || "" == FilterSubst;
```

Comment:
Here you're checking against an implicit default value for both options; a slightly more ergonomic way to do it is to pass a `cl::init` parameter to the initializer for each option.

Reply:
That's a good idea. I completely missed the cl::inti on llvm's documentation. Thanks for pointing that out!